app/controllers/architectures_controller.rb

```diff
@@ -39,4 +56,55 @@ def destroy
       process_error
     end
   end
+
+  private
+
+  def can_edit?(id)
+    authorized_for(controller: :architectures, action: :update, id: id)
+  end
+
+  def can_delete?(id)
+    authorized_for(controller: :architectures, action: :destroy, id: id)
+  end
+
+  def row_permissions(a)
+    { canEdit: can_edit?(a), canDelete: can_delete?(a) }
+  end
+
+  def columns
+    [
+      {
+        sortKey: 'name',
+        title: _('Name'),
+        width: 15,
```

Comment:
I'm a bit concerned with putting display hints in the data model. I would vote for adding those in the view layer.